make_hh_acs_data.R

```diff
@@ -836,58 +718,140 @@ get_acs_hs <- function(year) {
     mutate(year = year)
 }
 
-d_acs$acs_hs <-
-  mappp_dfr(2012:2021, get_acs_hs) |>
-  add_col_attrs(fraction_hs,
-    title = "Fraction of Adults with At Least High School Education",
-    description = "Available from 2012 onwards"
-  )
+d_acs$acs_hs <- mappp_dfr(2012:2022, get_acs_hs) 
 
 d <- purrr::reduce(d_acs, left_join, by = c("census_tract_id", "census_tract_vintage", "year"))
 
 d <- d |>
+  as_tibble() |>
   dplyr::relocate(c(census_tract_id, census_tract_vintage, year)) |>
   mutate(across(starts_with("fraction_"), \(.) round(., 3))) |>
   mutate(across(starts_with("n_"), as.integer)) |>
   mutate(across(starts_with("median_"), \(.) signif(., 3)))
 
 d <- d |>
-  add_col_attrs(census_tract_id,
-    title = "Census Tract Identifier",
-    description = "can refer to 2010 or 2020 vintage census tracts; unique only in combination with `census_tract_vintage`"
-  ) |>
-  add_col_attrs(census_tract_vintage,
-    title = "Census Tract Vintage",
-    description = "The year of the decennial census that defines the tract (2010 for years 2010-2019 and 2020 for years 2020-2029)"
-  ) |>
-  add_col_attrs(year,
-    title = "Year",
-    description = "The year of the 5-year ACS estimates (e.g., the 2019 ACS covers 2015 - 2019)"
-  )
-
-d <- d |>
-  add_attrs(
+  as_fr_tdr(
     name = "hh_acs_measures",
-    version = "1.1.0",
+    version = "1.2.0",
     title = "Harmonized Historical American Community Survey Measures",
-    description = "2010 - 2021 measures derived from ACS variables for census tracts in the contiguous US",
-    homepage = "https://geomarker.io/hh_acs_measures",
-  ) |>
-  add_type_attrs()
+    description = "2010 - 2022 measures derived from ACS variables for census tracts in the contiguous US",
+    homepage = "https://geomarker.io/hh_acs_measures"
+    ) |>
+  update_field("census_tract_id",
+               title = "Census Tract Identifier",
+               description = "can refer to 2010 or 2020 vintage census tracts; unique only in combination with `census_tract_vintage`"
+               ) |>
+  update_field("census_tract_vintage",
+               title = "Census Tract Vintage",
+               description = "The year of the decennial census that defines the tract (2010 for years 2010-2019 and 2020 for years 2020-2029)"
+               ) |>
+  update_field("year",
+               title = "Year",
+               description = "The year of the 5-year ACS estimates (e.g., the 2019 ACS covers 2015 - 2019)"
```

Comment:
What was the issue with including metadata alongside data calculation?  I like keeping them close together in the script so that we can verify documentation right beside the field as it is created.

Maybe we can fix the issue or make it better with `fr` somehow?

Reply:
I was converting back and forth from `fr_tdr` to `tibble` and losing the metadata in the conversions, but I forgot about the `dpylr`-like `fr` functions, so I think I can change it back.

Reply:
actually I am losing all the metadata (except for the first variable `fraction_poverty`) at the `purr::reduce()` step.